VK_App/VK_App/ViewControllers/FirstBarVC/FriendsTableViewController.swift

```diff
@@ -41,116 +21,128 @@ final class FriendsTableViewController: UITableViewController {
-        User(
-            friendName: Constants.secondName,
-            friendImage: Constants.secondImageName
-        ),
-        User(
-            friendName: Constants.thirdName,
-            friendImage: Constants.thirdImageName
-        ),
-        User(friendName: Constants.fourthName, friendImage: Constants.fourthImageName),
-        User(friendName: Constants.fiveName, friendImage: Constants.fiveImageName),
-        User(friendName: Constants.sixName, friendImage: Constants.sixImageName),
-        User(
-            friendName: Constants.sevenName,
-            friendImage: Constants.sevenImageName,
-            userPhotoNames: [Constants.firstLizaImageName, Constants.secondLizaImageName, Constants.thirdLizaImageName]
-        ),
-    ]
-
-    private var filteredFriendsMap: [Character: [(String, String, [String]?)]] = [:]
-
     private let networkAPIService = NetworkAPIService()
+    private var sectionsMap: [Character: [User]] = [:]
+    private var users: [User] = [] {
+        didSet {
+            filteredFriendsMap = [:]
+            sortedMethod()
+        }
+    }
+
+    private var filteredFriendsMap: [Character: [User]] = [:] {
+        didSet {
+            tableView.reloadData()
+        }
+    }
+
+    private var sortedCharacters: [Character] {
+        filteredFriendsMap.keys.sorted()
+    }
 
     // MARK: Life cycle
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        searchBarDelegate()
-        sortedMethod()
-        infoAboutUser()
+        setMethods()
     }
 
     // MARK: Private Methods
 
-    private func infoAboutUser() {
-        networkAPIService.fetchFriends()
-        networkAPIService.fetchUserPhotos()
-        networkAPIService.fetchGroups()
-        networkAPIService.fetchSearchedGroup(group: Constants.searchGroupName)
+    private func setMethods() {
+        searchBarDelegate()
+        fetchFriends()
+    }
+
+    private func fetchFriends() {
+        networkAPIService.fetchFriends { [weak self] result in
+            switch result {
+            case let .success(users):
+                DispatchQueue.main.async {
+                    self?.users = users
+                    self?.tableView.reloadData()
+                }
+            case let .failure(error):
+                print(error.localizedDescription)
+            }
+        }
     }
 
     private func searchBarDelegate() {
         searchBar.delegate = self
     }
 
     private func sortedMethod() {
-        for users in friends {
-            guard let firstChar = users.friendName.first
-            else { return }
-
-            if sectionsMap[firstChar] != nil {
-                sectionsMap[firstChar]?.append((users.friendName, users.friendImage, users.userPhotoNames))
+        var results = [Character: [User]]()
+        users.forEach {
+            guard let character = $0.firstName.first else { return }
+            if results[character] != nil {
+                results[character]?.append($0)
```

Comment:
```suggestion
            guard let character = $0.firstName.first ,
            results[character] != nil else { 
            results[character] = [$0]
            return }
                results[character]?.append($0)
```

81- 83  строки удалить